aiohttp/client_reqrep.py

```diff
@@ -284,7 +284,11 @@ def _writer(self, writer: Optional["asyncio.Task[None]"]) -> None:
             self.__writer.remove_done_callback(self.__reset_writer)
         self.__writer = writer
         if writer is not None:
```

Comment:
Could you reduce nesting by flipping this and bailing early + doing something similar to the block below?

Reply:
done in aee68303